### DISCUSSION TITLE
Stopping Code

### DISCUSSION THREAD

**Discussion by SaharshSS** [pybricks/support#680] [Q&A]
Hi, I am running a simple code for a four speed gearbox on the PU city hub. For some reason, the stop button in pybricks doesn't work. How do I get it to work?

**Comment by laurensvalk** [comment]
What code are you trying to run?

**Reply by SaharshSS** [reply]
Just a simple program running a medium linear motor in one direction.

**Reply by laurensvalk** [reply]
Sharing the code usually helps us diagnose the problem, and fix it if we can.

For example, maybe the stop button does work to stop the program, but not the motor. Then we’ll know to look for the problem in the motor driver instead of the button driver.
